### ISSUE TITLE
Динамический график USD/RUB + ФКТ/СВОДД + Список комментариев

### ISSUE THREAD

**Issue by audetv** [audetv/fct-search#61]
Концепция этой схемы заключается в том, чтобы прослеживать связи между попадающими в тему новостями и общим ходом обсуждения, одновременно имея в виду как объём комментариев за период, так и ход изменений курса фантиков. 

Примерный вид:
https://cont.ws/uploads/pic/2023/4/fct_graph_comment_usd.png

Масштабируемый график «японских свечей» USD/RUB, данные берутся с ММВБ, например:
https://www.moex.com/ru/issue/USD000UTSTOM/CETS

Спецификации данных ММВБ:
https://www.moex.com/ru/orders?historicaldata

График ФКТ/СВОДД выполнен по той же модели, что на главной странице сайта, только меняем у него окно времени. Сейчас столбец привязан к теме, а в новом графике столбец привязан ко времени. Окно времени задаётся в зависимости от разрешения графика валют на ММВБ. На бирже много разрешений, нам столько не нужно, хватит двух или трёх, так, чтобы на экране умещались 30-50 столбцов. Например варианты:
Масштаб 1 — график включает данные за месяц, каждая свеча представляет собой один день.
Масштаб 2 — график данных за неделю, каждая свеча представляет собой 5-6 часов.
Масштаб 3 — график данных за сутки, каждая свеча 15-20 минут.
Масштаб 4 — полный график с начала СВОДД, одна свеча = одна неделя. 

Под графиками идёт основное — список комментариев в фокусе центра «окна времени». То есть, при выбранном масштабе можно кликнуть по свече и внизу отобразятся комментарии, сделанные во время неё. 

Примерная схема работы может варьироваться. Поиграть со значениями окна времени можно на сайте ММВБ или на других площадках типа таких: 
https://ru.investing.com/currencies/usd-rub-chart
https://ru.tradingview.com/

В списке комментариев можно либо показывать все комментарии сайта, либо только из темы СВОДД.

**Comment by audetv** [comment]
Идея понятна, надо изучить техническую часть, я немного днем посмотрел по ссылкам и почитал и пока нет понимания как это сделать. Надо будет углубиться в вопрос.

**Comment by iprst** [comment]
График курса не обязательно японскими свечами (это наглядно, но ресурсоёмко), можно отобразить его кривой, как это сделано во многих приложениях. Сама сцепка столбцов японских свечей и графика комментариев будет очень наглядной, можно будет отслеживать тенденции. При этом график ФКТ займёт обычное место объёма торгов, как например на графике ММВБ по ссылке ранее — обычно это стандартная реализация выносного индикатора на биржевом графике, и скорее всего существуют готовые инструменты. В дальнейшем по желанию такую схему можно будет расширить, включив наряду с комментариями парсер новостей или других источников данных.

**Comment by audetv** [comment]
Нужна помощь с тем, откуда брать информацию о торгах, бесплатно, но можно с регистрацией, источники. Список ссылок, где посмотреть. Я сегодня гуглю это вопрос и пока не всегда понимаю, что я нахожу и как с этим работать, пока буду добавлять ссылки.

https://iss.moex.com/iss/reference/
https://iss.moex.com/iss/statistics/engines/currency/markets/selt/rates.json?iss.meta=off

> Спецификации данных ММВБ:
> https://www.moex.com/ru/orders?historicaldata

Найти бы архивные данные, но бесплатно. Я предполагаю в реальном времени, условно после того, как запустить программу сборщик-отслеживания курса, можно собирать и записывать данные, но нам же интересны и прошлые периоды тоже.

**Comment by iprst** [comment]
Интересующий нас инструмент называется USDRUB_TOM, это валютная пара доллар США – российский рубль, торгующаяся на ММВБ, где TOM означает, что срок исполнения обязательств по сделке переходит на первый рабочий день после ее проведения; либо. Получить данные можно многими способами, например для всяких дестопных программ его обычно скачивают с ФИНАМа:
https://www.finam.ru/profile/mosbirzha-valyutnyj-rynok/usdrubtom-usd-rub/export/

Там будет txt / csv с заданным разделителем. 

Архивные данные называются историческими и скачиваются там же — нужно задать окно времени, шаг свечи и так далее.

**Comment by iprst** [comment]
> С 10:00 до 18:00 запросы сделок больше чем за один день запрещены

Это касается только малых размерностей времени, например тиков. Часовики скачиваются спокойно, для проверки скачал за период 05.04.2019 — 13.04.2023, всё в порядке, файл формируется. 

[USD000UTSTOM_190405_230413.txt](https://github.com/audetv/fct-search/files/11223411/USD000UTSTOM_190405_230413.txt)

**Comment by ZelAOYury** [comment]
Добрый день. Вроде сюда попал. Если буду тупить, то подсказывайте, как тут всё устроено.

**Comment by audetv** [comment]
> Добрый день. Вроде сюда попал. Если буду тупить, то подсказывайте, как тут всё устроено.

Юрий, добро пожаловать, подскажем, покажем, присоединяйтесь к любому обсуждению, можно свободно создавать новые issue (задачи, темы) зеленая кнопка «New Issue» или общаться в уже созданных.

**Comment by ZelAOYury** [comment]
А есть какая то общая, куда идеи скидывать? А потом интересные можно в отдельную выделять.

**Comment by iprst** [comment]
> А есть какая то общая, куда идеи скидывать? А потом интересные можно в отдельную выделять.

Я скидывал сюда, в первую вкладку:
https://github.com/users/audetv/projects/6, 

Но Алексею в таком случае приходится вручную делать из идей Issue чтобы можно было обсуждать и дополнять, поэтому лучше как он рекомендует выше, сразу писать в «проблемы»:
https://github.com/audetv/fct-search/issues

Там надо сделать понятный короткий заголовок, а текст описания можно любой. Потому что когда накопится много, нужно чтобы было понятно из заголовка о чём речь.

**Comment by audetv** [comment]
Во тут мы обсуждали общую концепцию поиска https://github.com/audetv/fct-search/issues/62 [Сервис поиск информации](https://github.com/audetv/fct-search/issues/62)
Можно там продолжать обсуждение, я эту тему закрепил pin сверху в списке всех issue. Наверное, можно ее переименовать, предлагайте)
Или можно создать новую тему «общее обсуждение», я из любой темы могу создать новый issue и далее отработать в коде, т.е. превратить чисто в технический вопрос, выгружается код и далее закрывается или открывается заново. Это удобно при разработке. из комментария делаю issue и работаю.

**Comment by iprst** [comment]
> «общее обсуждение»

Так и до гуглобуса дойдём!

**Comment by audetv** [comment]
> «общее обсуждение»
> 
> Так и до гуглобуса дойдём!
Так точно!

Прямо перед глазами была ссылка, только я не пользовался этим, похоже это то, что нам нужно: 
https://github.com/audetv/fct-search/discussions
Я пока открыл шаблонную тему объявление с текстом по умолчанию, надо будет текст изменить или сделать перевод,.

**Comment by audetv** [comment]
> Прямо перед глазами была ссылка, только я не пользовался этим, похоже это то, что нам нужно:

Не было ссылки перед глазами пока я не зашел в настройки не включил в секции Features галочку дискуссий.
Сделал и забыл... только позже осознал, что сам включил.

**Comment by iprst** [comment]
С ФКТ летит совет: 

Для реализации маркеров курс/новости, попробуйте виджеты https://www.tradingview.com/widget/

**Comment by audetv** [comment]
Да, тоже видел, хорошо, надо будет посмотреть, подходит ли нам это.
Пока, по диагонали прочитал, понял что там встраивается их график с их данными на сайт.
https://www.tradingview.com/widget/advanced-chart/
Вроде встраиваемый iframe. На работе блокируется сайт, через мобильный только удалось пока посмотреть.

**Comment by iprst** [comment]
Имеет смысл встраивать свой график, функционал которого будет небольшим, но конкретным. Этот функционал ещё нужно продумать в деталях, пока только идея.
